relay-server/src/actors/outcome.rs

```diff
@@ -622,31 +711,230 @@ impl Actor for HttpOutcomeProducer {
     type Context = Context<Self>;
 }
 
-impl Supervised for HttpOutcomeProducer {}
+impl Handler<TrackRawOutcome> for HttpOutcomeProducer {
+    type Result = Result<(), OutcomeError>;
+
+    fn handle(&mut self, msg: TrackRawOutcome, ctx: &mut Self::Context) -> Self::Result {
+        self.send_http_message(msg, ctx);
+        Ok(())
+    }
+}
+
+impl Handler<TrackOutcome> for HttpOutcomeProducer {
+    type Result = Result<(), OutcomeError>;
+
+    fn handle(&mut self, msg: TrackOutcome, ctx: &mut Self::Context) -> Self::Result {
+        self.handle(TrackRawOutcome::from_outcome(msg, &self.config), ctx)
+    }
+}
+
+struct ClientReportOutcomeProducer {
+    flush_interval: Duration,
+    unsent_reports: BTreeMap<Scoping, Vec<ClientReport>>,
+}
+
+impl ClientReportOutcomeProducer {
+    fn create(config: &Config) -> Self {
+        Self {
+            // Use same batch interval as outcome aggregator
+            flush_interval: Duration::from_secs(config.outcome_aggregator().flush_interval),
+            unsent_reports: BTreeMap::new(),
+        }
+    }
+
+    fn flush(&mut self, _ctx: &mut Context<Self>) {
+        relay_log::trace!("Flushing client reports");
+        let unsent_reports = mem::take(&mut self.unsent_reports);
+        let envelope_manager = EnvelopeManager::from_registry();
+        for (scoping, client_reports) in unsent_reports.into_iter() {
+            envelope_manager.do_send(SendClientReports {
+                client_reports,
+                scoping,
+            });
+        }
+    }
+}
+
+impl Actor for ClientReportOutcomeProducer {
+    type Context = Context<Self>;
+
+    fn started(&mut self, ctx: &mut Self::Context) {
+        ctx.run_interval(self.flush_interval, Self::flush);
```

Comment:
@untitaker In the end I decided to use `run_interval` here. The outcomes for `ClientReportOutcomeProducer` always come from the aggregator in predictable intervals, so I suppose there is no need for "flush when batch is full or too much time has passed" logic. Instead, just flush with the same interval as the aggregator.